pudb/lowlevel.py

```diff
@@ -28,6 +28,37 @@
 
 from pudb.py3compat import PY3, text_type
 
+import logging
+
+
+class SafeStreamHandler(logging.StreamHandler):
+    """
+    Logging stream handler that clears up the UI when it's done, if needed.
+    """
+    def emit(self, record):
+        from pudb import _have_debugger, _get_debugger
+        super(SafeStreamHandler, self).emit(record)
+        if _have_debugger():
+            dbg = _get_debugger()
+            dbg.ui.screen.clear()
```

Comment:
Does this wait for the user to see the error message? If not, would it make sense to add that? Alternatively, given that we have a way of getting to the `DebuggerUI` object from here, we could simply call `DebuggerUI.message()` with the content of the message.

Reply:
I'd rather not do that, since the log fires quite regularly when there's a problem in the variable view- once per refresh in fact, if memory serves, so requiring user input every time would render the session unusable. We'd have to figure out a way of caching prior messages and only logging them the first time, which could be possible I suppose, but might also make it harder to figure out why a message is appearing, since it won't reappear.

Reply:
That's a fair point. As an alternative, we could direct those messages to pudb's built-in console (which we would have to force-show in the event of output to make sure the messages are seen by the user). How does that sound?

Reply:
I could go with that. I'd want to still send them to stderr as well, or perhaps provide an option to send them to a file instead.

Reply:
That sounds reasonable. Thanks for being receptive to the suggestion.